app/components/Modals/SendModal/SendModal.jsx

```diff
@@ -4,91 +4,43 @@ import React from 'react'
 import BaseModal from '../BaseModal'
 import ReadCode from './ReadCode'
 import ConfirmDetails from './ConfirmDetails'
-
-import parseQRCode from '../../../util/parseQRCode'
+import type { RecipientData } from '../../../util/parseQRCode'
 
 type Props = {
   hideModal: () => any,
-  showErrorNotification: (error: Object) => any,
-  hideNotification: (id: string) => any,
-  pushQRCodeData: (data: Object) => any
-}
-
-type State = {
-  step: string,
-  error: ?string,
-  recipientData: Object
+  pushQRCodeData: (data: Object) => any,
+  getRecipientData: string => any,
+  clearRecipientData: Function,
+  recipientData: ?RecipientData
 }
 
-export default class SendModal extends React.Component<Props, State> {
-  state = {
-    step: '1',
-    error: null,
-    recipientData: {}
-  }
-
-  displayError = (message: string) => {
-    const { showErrorNotification } = this.props
-
-    const newError = showErrorNotification({
-      message: `An error occurred while scanning this QR code: ${message}. Please try again.`
-    })
-
-    this.setState({ error: newError })
-  }
-
-  isStepTwo = () => this.state.step === '2'
-
-  gotoPreviousStep = () => {
-    this.setState({
-      step: '1'
-    })
-  }
-
-  gotoNextStep = (recipientData: string, stopScanner: Function) => {
-    const { error } = this.state
-    const { hideNotification } = this.props
-
-    try {
-      const parsedRecipientData = parseQRCode(recipientData)
-
-      stopScanner()
-
-      if (error) hideNotification(error)
-
-      this.setState({
-        step: '2',
-        recipientData: parsedRecipientData
-      })
-    } catch (message) {
-      this.displayError(message)
-    }
-  }
-
-  confirmAndClose = () => {
+export default class SendModal extends React.Component<Props> {
+  confirmAndClose = (recipientData: RecipientData) => {
     const { pushQRCodeData, hideModal } = this.props
 
-    pushQRCodeData(this.state.recipientData)
+    pushQRCodeData(recipientData)
     hideModal()
   }
 
-  getStepComponent = () =>
-    ({
-      '1': <ReadCode gotoNextStep={this.gotoNextStep} />,
-      '2': (
-        <ConfirmDetails
-          recipientData={this.state.recipientData}
-          confirmAndClose={this.confirmAndClose}
-        />
-      )
-    }[this.state.step])
+  getStepComponent = () => {
```

Comment:
@ranbena perhaps in a preference, but this would actually look pretty nice with a class getter

`get stepComponent() { .. }`

then inside the render function:

`this.stepComponent`

WDYT?

Reply:
Sure! a217283